### PR TITLE
reference: reana.yaml updates

### LINKED ISSUE #21: docs: reana.yaml
Write basic docs about  ``reana.yaml`` syntax, meaning, allowed fields, validation.

### PR DESCRIPTION
**Pull request by tiborsimko**
* Expands Reference section guide on `reana.yaml` using more technical
  language. (closes #21)

Signed-off-by: Tibor Šimko <tibor.simko@cern.ch>

### CODE REVIEW

File: docs/reference/reana-yaml/index.md [hunk 1]
```diff
@@ -1,112 +1,219 @@
...
 
-To be added
+The **workflow.specification.steps** property describes each individual computational step of the Serial workflow and is further composed of:
 
-### YADAGE
+| Property | Type   | Mandatory?  | Description                                        |
+| -------- | ----   | ----------  | -----------                                        |
+| name    | *string* | *optional* | Provides name of the given workflow step. |
+| environment  | *string*  | *mandatory*  | Specifies runtime environment container image where the given workflow step commands will be run.  |
+| commands  | *list*  | *mandatory*  | Lists all commands to be run in the runtime environment container image when the given workflow step is executed.  |
```
Comment (mvidalgarcia) [inline docs/reference/reana-yaml/index.md]: Perhaps here we can mention that each command will be run in a separate runtime container? Unless we change it. IIRC this has caused some confusion during the Awesome Workshop.

File: docs/reference/reana-yaml/index.md [hunk 2]
```diff
@@ -1,112 +1,219 @@
...
+A more detailed information on workflow specification languages is available in
+corresponding [CWL](/running-workflows/supported-systems/cwl),
+[Serial](/running-workflows/supported-systems/serial) and
+[Yadage](/running-workflows/supported-systems/yadage) pages.
+
+### reana.yaml outputs
+
+The **outputs** property of `reana.yaml` specifies all the workflow high-level
+outputs, consisting of a list of files. The `reana-client download` command
+will download all the specified files from to the workflow workspace to the
```
Comment (mvidalgarcia) [inline docs/reference/reana-yaml/index.md]:
```suggestion
will download all the specified files from the workflow workspace to the
```

File: docs/reference/reana-yaml/index.md [hunk 3]
```diff
@@ -1,112 +1,219 @@
...
-   $ reana-client validate
-   File /Users/johndoe/my-analysis/reana.yaml is a valid REANA specification file.
+$ reana-client validate
+File my-analysis/reana.yaml is a valid REANA specification file.
 ```
 
-If your workflow specification file is not named `reana.yaml` or `reana.yml` you have
-to use `-f` flag in order to specify a file path for the validation.
+If your workflow specification file is not named `reana.yaml` (or `reana.yml`),
+you can use the to use `-f` command line option to specify path to the file for
```
Comment (mvidalgarcia) [inline docs/reference/reana-yaml/index.md]:
```suggestion
you can use the `-f` command-line option to specify the path to the file for
```

File: docs/reference/reana-yaml/index.md [hunk 4]
```diff
@@ -1,112 +1,219 @@
...
+
+The **output** property is composed of:
+
+| Property    | Type         | Mandatory? | Description                                 |
+| --------    | ----         | ---------- | -----------                                 |
+| files       | *list*       | *optional* | Lists all the output files of the workflow. |
+
+The **outputs** property example:
+
+```yam
```
Comment (mvidalgarcia) [inline docs/reference/reana-yaml/index.md]:
```suggestion
```yaml
```

File: docs/reference/reana-yaml/index.md (hunk 1 above)
Comment (tiborsimko) [reply docs/reference/reana-yaml/index.md]: Added a brief note, but we'll still have to explain in detail in the overall Serial documentation.

**Pull request merged**